app/src/main/java/com/example/drawingApp/ColorPickerUtility.kt

```diff
@@ -0,0 +1,222 @@
+package com.example.drawingApp
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.graphics.Color
+import android.view.LayoutInflater
+import android.widget.TextView
+import androidx.core.graphics.toColor
+import androidx.core.widget.addTextChangedListener
+import com.google.android.material.bottomsheet.BottomSheetDialog
+import kotlin.math.max
+import kotlin.math.min
+
+object ColorPickerUtility {
+
+    @SuppressLint("ClickableViewAccessibility")
+    fun colorPickerSheet(
+        context: Context,
+        /**
+         * Called whenever the hue is updated via text or slider.
+         * Use it to set wherever you are storing your hue
+         * and then update your color
+         */
+        onHueUpdate: (Float) -> Unit,
+        /**
+         * Called whenever the Saturation or Value is updated via the gradient square
+         * or via text. use it to set wherever you are storing your saturation AND value
+         * update your color afterward
+         */
+        onSatOrValUpdate: (Float, Float) -> Unit,
+        /**
+         * Called whenever text needs to be updated, meaning on any change, hsv, rgb, or text.
+         * Used to give the information of hsv and previous color.
+         * if there was no previous color pass it black. This is used to help
+         * populate the sheets text views and update the slider and gradient indicators as well as set the previous color box.
+         */
+        onColorTextForceUpdate: () -> DrawingViewModel.ColorPack,
```

Comment:
I tend to prefer using a separate lambda/value to provide a previous value.

I noted photoshop defaults to Red on startup, but that's just a random preference, any color should be fine.

Also, I asked Nic about it, but you might want to consider putting the data class in a ColorUtil file and instead only holding an array or hex string in the viewModel.

Reply:
putting the data class in the util file does make sense